pkg/security/secl/model/process_cache_entry_unix.go

```diff
@@ -15,26 +15,22 @@ import (
 
 // SetAncestor sets the ancestor
 func (pc *ProcessCacheEntry) SetAncestor(parent *ProcessCacheEntry) {
-	if pc.Ancestor == parent {
+	if pc.Ancestor == &parent.ProcessContext {
 		return
 	}
 
-	if pc.Ancestor != nil {
-		pc.Ancestor.Release()
-	}
-
-	pc.Ancestor = parent
+	pc.Ancestor = &parent.ProcessContext
```

Comment:
I'm a bit concern about not releasing the ancestor, this could lead into a memory leak